src/API/AbstractAPIClient.php

```diff
@@ -51,9 +53,18 @@ public function callAPI(Request $request)
                 $requestOptions['debug'] = fopen('php://stderr', 'w');
             }
 
-            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $requestOptions);
+            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $request->getHeaders(), $request->getBody(), $request->getParameters());
 
-            $response = $apiRequest->send($apiRequest)->json();
+            $method = $request->getMethod();
+
+            // Since Guzzle automatically overwrites a new header when the request
+            // is POST / PATCH / DELETE, we need to overwrite the Content-Type header
+            // with setBody() function.
+            if ($method === 'PATCH' || $method === 'DELETE' || $method === 'POST') {
```

Comment:
I'm not certain. Except for post nothing is documented. I had to discover PATCH and DELETE myself. To be honest we can just override the Content-Type to `'application/json'` in all cases.